src/pages/HomePage/HomePage.tsx

```diff
@@ -80,52 +73,52 @@ export const HomePage = () => {
     navigate(`/${newPage}${idParam ? `/${idParam}` : ''}`);
   };
 
+  const handleRefresh = () => {
+    if (isSearch) {
+      searchRefetch();
+    } else {
+      latestRefetch();
+    }
+  };
+
   return (
     <div className="flex-grow">
+      {loading && <Loader />}
       <SearchBar onSearch={handleSearch} initialValue={searchTerm} />
 
-      <div className="flex flex-col md:flex-row gap-6 transition-all items-stretch min-h-[400px]">
+      <div className="flex min-h-[400px] flex-col items-stretch gap-6 transition-all md:flex-row">
         <div className={`${idParam ? 'md:w-2/3' : 'w-full'} h-full`}>
           {error && (
-            <div className="text-center text-red-600 font-medium mb-4">
+            <div className="mb-4 text-center font-medium text-red-600">
               {error}
             </div>
           )}
           <CardList items={images} />
         </div>
 
         {idParam && (
-          <div className="w-full md:w-1/3 h-full">
+          <div className="h-full w-full md:w-1/3">
             <Outlet />
           </div>
         )}
       </div>
 
-      <div className="flex justify-center items-center gap-4 mt-8">
-        <button
-          data-testid="prev-btn"
+      <div className="mt-8 flex items-center justify-center gap-4">
+        <AppButton
+          text="Previous"
           onClick={() => goToPage(page - 1)}
           disabled={page <= Number(DEFAULT_PAGE)}
-          className="px-4 py-2 bg-slate-600 text-white rounded hover:bg-slate-700 disabled:opacity-50 disabled:cursor-not-allowed cursor-pointer"
-        >
-          Previous
-        </button>
+        />
 
-        <span className="text-slate-700 dark:text-gray-100 font-medium px-2">
+        <span className="px-2 font-medium text-slate-700 dark:text-gray-100">
           Page {page}
           {totalPages ? ` of ${totalPages}` : ''}
         </span>
 
-        <button
-          data-testid="next-btn"
-          onClick={() => goToPage(page + 1)}
-          className="px-4 py-2 bg-slate-600 text-white rounded hover:bg-slate-700 cursor-pointer"
-        >
-          Next
-        </button>
+        <AppButton text="Next" onClick={() => goToPage(page + 1)} />
```

Comment:
Does next button take into account last page? It has to be disabled on the last page